core/src/test/java/org/elasticsearch/ingest/core/CompoundProcessorTests.java

```diff
@@ -117,4 +111,68 @@ public void testSingleProcessorWithNestedFailures() throws Exception {
         assertThat(processorToFail.getInvokedCounter(), equalTo(1));
         assertThat(lastProcessor.getInvokedCounter(), equalTo(1));
     }
+
+    public void testCompoundProcessorExceptionFailWithoutOnFailure() throws Exception {
```

Comment:
I'm not sure this method name is right. Maybe I just don't know this code that well though.